core/aws_ddk_core/stages/s3_event.py

```diff
@@ -74,18 +77,27 @@ def __init__(
             },
         )
 
-        self._trail_bucket = S3Factory.bucket(
-            self,
-            id=f"{id}-trail-bucket",
-            environment_id=environment_id,
+        self._trail_bucket: Optional[IBucket] = (
```

Comment:
Nitpick - would it be cleaner to have a singe `if not cloudtrail:` and create bucket and trail?